test/suites/etcd.robot

```diff
@@ -0,0 +1,80 @@
+*** Settings ***
+Documentation       Tests related to how etcd is managed
+
+Resource            ../resources/common.resource
+Resource            ../resources/systemd.resource
+Resource            ../resources/microshift-config.resource
+Resource            ../resources/microshift-process.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           etcd  configuration  restart  slow
+
+
+*** Variables ***
+${ETCD_SYSTEMD_UNIT}    microshift-etcd.scope
+${MEMLIMIT128}      SEPARATOR=\n
+...  ---
+...  etcd:
+...  \ \ memoryLimitMB: 128
+${MEMLIMIT0}      SEPARATOR=\n
+...  ---
+...  etcd:
+...  \ \ memoryLimitMB: 0
+
+
+*** Test Cases ***
+Set MemoryHigh Limit Unlimited
+    [Documentation]  The default configuration should not limit RAM
+    ...
+    ...  Since we cannot assume that the default configuration file is
+    ...  being used, the test explicitly configures a '0' limit, which
+    ...  is equivalent to not having any configuration at all.
+    [Setup]  Setup With Custom Config  ${MEMLIMIT0}
+    Expect MemoryHigh  infinity
+
+Set MemoryHigh Limit 128MB
+    [Documentation]  Set the memory limit for etcd to 128MB and ensure it takes effect
+    [Setup]  Setup With Custom Config  ${MEMLIMIT128}
+    # Expecting the setting to be 128 * 1024 * 1024
+    Expect MemoryHigh  134217728
+    [Teardown]  Restore Default Config
+
+
+*** Keywords ***
+Setup
+    [Documentation]  Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig  # for readiness checks
+    Save Default MicroShift Config
+
+Teardown
+    [Documentation]  Test suite teardown
+    Restore Default Config
+    Logout MicroShift Host
+    Remove Kubeconfig
+
+Restore Default Config
+    [Documentation]  Remove any custom config and restart MicroShift
+    Restore Default MicroShift Config
+    Restart MicroShift
+    Sleep  10 seconds  # Wait for systemd to catch up
+
+Setup With Custom Config
+    [Documentation]  Install a custom config and restart MicroShift
+    [Arguments]  ${config_content}
+    ${merged}=  Extend MicroShift Config  ${config_content}
+    Upload MicroShift Config  ${merged}
+    Restart MicroShift
+    Sleep  10 seconds  # Wait for systemd to catch up
```

Comment:
The Restart MicroShift keyword is already waiting for all pods to be running, does systemd require more than that?

Reply:
It doesn't currently wait for pods, though it could in the future. I'll see if removing this makes the test less stable.

Reply:
Seems OK, I've removed it.